pkg/scaffold/v2/crd_viewer_rbac.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/gobuffalo/flect"
+
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/input"
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/v1/resource"
+)
+
+var _ input.File = &CRDViewerRole{}
+
+// CRD Viewer role scaffolds the config/rbca/<kind>_viewer_role.yaml
+type CRDViewerRole struct {
+	input.Input
+	// Plural is the plural lowercase of kind
+	Plural string
+
+	// Resource is a resource in the API group
+	Resource *resource.Resource
+}
+
+// GetInput implements input.File
+func (g *CRDViewerRole) GetInput() (input.Input, error) {
+	if g.Path == "" {
+		g.Path = filepath.Join("config", "rbac", fmt.Sprintf("%s_viewer_role.yaml", strings.ToLower(g.Resource.Kind)))
+	}
+	if g.Plural == "" {
+		g.Plural = flect.Pluralize(strings.ToLower(g.Resource.Kind))
+	}
+	g.TemplateBody = crdRoleViewerTemplate
+	return g.Input, nil
+}
+
+// Validate validates the values
+func (g *CRDViewerRole) Validate() error {
+	return g.Resource.Validate()
+}
+
+var crdRoleViewerTemplate = `# permissions to do viewer {{ .Resource.Kind }}.
```

Comment:
> permissions to do viewer {{ .Resource.Kind }}.

permissions to view {{ .Resource.Resource }}.

Reply:
Done 👍